docs/developer_guidelines.rst

```diff
@@ -41,9 +40,92 @@ Blocks:
 .. _pyflakes: https://pypi.python.org/pypi/pyflakes
 .. _should be grouped: https://www.python.org/dev/peps/pep-0008/#imports
 
+Code guidelines
+---------------
+Some guidelines to keep in mind when coding for Blocks. Some of these are simply
+preferences, others stem from particular requirements we have e.g. in order to
+serialize training progress, support Python 2 and 3 simultaneously, etc.
+
+Validating function arguments
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+In general, be Pythonic and use `duck typing`_. That is, don't check the type of
```

Comment:
Well, I think sometimes checks make a lot of sense. They prevent highly non-trivial bugs. I am not sure that explicitly encouraging people not to do checks (what they anyways tend not to do) is the right way to go for Blocks.